storage/database.go

```diff
@@ -134,6 +133,17 @@ func NewDatabase(shardSet sharding.ShardSet, opts Options) (Database, error) {
 		return nil, errCommitLogStrategyUnknown
 	}
 
+	ns := make(map[string]databaseNamespace, len(namespaces))
+	for _, n := range namespaces {
+		if _, exists := ns[n.Name()]; exists {
+			return nil, errDuplicateNamespaces
+		}
+		// NB(xichen): shardSet is used only for reads but not writes once created
+		// so can be shared by different namespaces
```

Comment:
shardSet is immutable, there should be no issue sharing a shardSet amongst two things no?  Not sure what this comment is getting at.